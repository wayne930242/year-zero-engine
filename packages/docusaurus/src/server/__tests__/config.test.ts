/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import path from 'path';
import loadConfig from '../config';

describe('loadConfig', () => {
  test('website with valid siteConfig', async () => {
    const fixtures = path.join(__dirname, '__fixtures__');
    const siteDir = path.join(fixtures, 'simple-site');
    const config = loadConfig(siteDir);
    expect(config).toMatchSnapshot();
    expect(config).not.toEqual({});
  });

  test('website with incomplete siteConfig', () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'bad-site');
    expect(() => {
      loadConfig(siteDir);
    }).toThrowErrorMatchingSnapshot();
  });

  test('website with useless field (wrong field) in siteConfig', () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'wrong-site');
    expect(() => {
      loadConfig(siteDir);
    }).toThrowErrorMatchingSnapshot();
  });

  test('website with no siteConfig', () => {
    const siteDir = path.join(__dirname, '__fixtures__', 'nonExisting');
    expect(() => {
      loadConfig(siteDir);
    }).toThrowErrorMatchingSnapshot();
  });
});
