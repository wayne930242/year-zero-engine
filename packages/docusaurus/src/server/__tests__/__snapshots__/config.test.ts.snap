// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadConfig website with incomplete siteConfig 1`] = `
"\\"favicon\\" is required
\\"url\\" is required
"
`;

exports[`loadConfig website with no siteConfig 1`] = `"docusaurus.config.js not found at packages/docusaurus/src/server/__tests__/__fixtures__/nonExisting/docusaurus.config.js"`;

exports[`loadConfig website with useless field (wrong field) in siteConfig 1`] = `
"\\"favicon\\" is required
These field(s) [\\"useLessField\\",] are not recognized in docusaurus.config.js.
If you still want these fields to be in your configuration, put them in the 'customFields' attribute.
See https://v2.docusaurus.io/docs/docusaurus.config.js/#customfields"
`;

exports[`loadConfig website with valid siteConfig 1`] = `
Object {
  "baseUrl": "/",
  "customFields": Object {},
  "favicon": "img/docusaurus.ico",
  "noIndex": false,
  "onBrokenLinks": "throw",
  "onBrokenMarkdownLinks": "warn",
  "onDuplicateRoutes": "warn",
  "organizationName": "endiliey",
  "plugins": Array [
    Array [
      "@docusaurus/plugin-content-docs",
      Object {
        "path": "../docs",
      },
    ],
    "@docusaurus/plugin-content-pages",
  ],
  "presets": Array [],
  "projectName": "hello",
  "tagline": "Hello World",
  "themeConfig": Object {},
  "themes": Array [],
  "title": "Hello",
  "titleDelimiter": "|",
  "url": "https://docusaurus.io",
}
`;
